src/se/vidstige/jadb/managers/PackageManager.java

```diff
@@ -49,24 +49,32 @@ private void verifyOperation(String operation, String target, String result) thr
 
     public void remove(RemoteFile file) throws IOException, JadbException {
         InputStream s = device.executeShell("rm", "-f", Bash.quote(file.getPath()));
-        Stream.readAll(s, Charset.forName("UTF-8"));
+        Stream.flushRead(s);
+        s.close();
     }
 
-    private void install(File apkFile, List<String> extraArguments) throws IOException, JadbException {
-        RemoteFile remote = new RemoteFile("/sdcard/tmp/" + apkFile.getName());
-        device.push(apkFile, remote);
+    private void install(String apkName, InputStream is, List<String> extraArguments) throws IOException, JadbException {
+        RemoteFile remote = new RemoteFile("/sdcard/tmp/" + apkName);
+        device.push(is, System.currentTimeMillis(), device.getMode(null), remote);
         ArrayList<String> arguments = new ArrayList<>();
         arguments.add("install");
         arguments.addAll(extraArguments);
         arguments.add(remote.getPath());
         InputStream s = device.executeShell("pm", arguments.toArray(new String[arguments.size()]));
         String result = Stream.readAll(s, Charset.forName("UTF-8"));
         remove(remote);
-        verifyOperation("install", apkFile.getName(), result);
+        verifyOperation("install", apkName, result);
+        is.close();
+        s.close();
+    }
```

Comment:
Revert changes to this method
* The apkFile is of type `File` to make it easier to understand what to pass, and also to avoid misstake. Don't chase bugs, the compiler can find! ;-)